### PR TITLE
Add error handling decorator for functions utilizing variable urls

### PR DESCRIPTION
**Pull request by adamhb123**
Add an error handling decorator for use with functions utilizing
variable urls, and apply that decorator to said functions. The original
issue states that this is a problem with /user/, but it is also an
issue with all functions that are routed to with variable urls.

Resolves: #78

### CODE REVIEW

**Comment by devinmatte** [comment]
Have you measured performance impact? I'm wondering if this ends up running the query twice

**Comment by adamhb123** [comment]
As far as page-loading speed goes, I'm not seeing any significant variation in performance, and I'm not seeing any signs of repeating any querys. Console output between the decorated and nondecorated versions of the functions seems exactly the same, and tossing a print statement into the user function only outputs to console once.

**Comment by mxmeinhold** [comment]
Did you consider an error handler rather than decorators? Flask can do the "wrap everything" part for you. 
https://flask.palletsprojects.com/en/1.1.x/errorhandling/#generic-exception-handlers

File: profiles/utils.py [hunk 1]
```diff
@@ -20,6 +20,24 @@
...
+    :return: wrapped function
+    """
+    @wraps(func)
+    def wrapped_function(*args, **kwargs):
+        try:
+            result = func(*args, **kwargs)
+            return result
+        #   pylint: disable=bare-except
+        except:
+            abort(404)
```
Comment (mxmeinhold) [inline profiles/utils.py:38]: I'm not sure I like returning a 404 for _any_ exception. Will this hide exceptions from sentry?

Comment (devinmatte) [reply profiles/utils.py:38]: I think it will. Caught exceptions won't bubble up

Comment (mxmeinhold) [reply profiles/utils.py:38]: Then @adamhb123 you should consider this requested changes, since this will kill a lot of our error tracking

Comment (adamhb123) [reply profiles/utils.py:38]: For sure I will, I've just been busy as of late. Will take a look sometime within the next few days.

**Comment by adamhb123** [comment]
Took a crack at it from a different angle, adding in a little 404 page as well. Tested and all seems to work well at least as far as the things I tossed at it go.

**Review by devinmatte — APPROVED** [review]
This is a much better solution, let's try it in dev

**Review by mxmeinhold — APPROVED** [review]
LGTM! :ship: :it:

**Pull request merged**